plugins/http_proxy/handler.go

```diff
@@ -21,23 +23,28 @@ type httpProxy struct {
 }
 
 func (h *httpProxy) Start(config config.HandlerConfig) (err error) {
-	options := loadFromConfig(config.Options)
-	addr := fmt.Sprintf("%s:%d", config.ListenAddress, config.Port)
-	h.server = &http.Server{Addr: addr, Handler: h.proxy}
+	var opts httpProxyOptions
+	if err = config.Options.Unmarshal(&opts); err != nil {
+		return
+	}
+	h.server = &http.Server{Addr: config.ListenAddr(), Handler: h.proxy}
```

Comment:
probably capture `config.ListenAddr()` in variable to avoid multiple formattings